### PR TITLE
darshan-runtime: Fix log path issue

### PR DESCRIPTION
**Pull request by sergiorg-hpc**

### CODE REVIEW

**Comment by sergiorg-hpc** [comment]
@pramodk It seems that the change worked. The idea of this PR is to force the log path of Darshan to be `~` instead of the `$HOME` variable when the module is being generated. The `~` will now be then interpreted after running `module load darshan-runtime`.

Here is the module configuration of the module deployed on BB5:
```
setenv DARSHAN_LOG_DIR_PATH "/gpfs/bbp.cscs.ch/ssd/gitlab_map_jobs//bbpcihpcdeploy/J324015"
```
Here is the new one:
```
setenv DARSHAN_LOG_DIR_PATH "~"
```
And now here is the result:
```
$ echo $DARSHAN_LOG_DIR_PATH 
/gpfs/bbp.cscs.ch/home/srivas
```

Simple :).

**Comment by sergiorg-hpc** [comment]
I’m not sure why the v3.4.0 is not built. @matz-e, any ideas or suggestions?

File: bluebrain/repo-patches/packages/darshan-runtime/package.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+    version("3.4.0", sha256="7cc88b7c130ec3b574f6b73c63c3c05deec67b1350245de6d39ca91d4cff0842", preferred=True)
+
+    depends_on('autoconf', type='build', when='@3.4.0:')
+    depends_on('automake', type='build', when='@3.4.0:')
+    depends_on('libtool',  type='build', when='@3.4.0:')
+    depends_on('m4',       type='build', when='@3.4.0:')
+
+    def setup_run_environment(self, env):
+        env.set('DARSHAN_LOG_DIR_PATH', "~")
```
Comment (sergiorg-hpc) [inline bluebrain/repo-patches/packages/darshan-runtime/package.py:11]: I will add a comment to clarify before merging, but first I want to understand why 3.4.0 is not building. I cannot see anything in our deployment specifications.

File: bluebrain/repo-patches/packages/darshan-runtime/package.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+from spack import *
+from spack.pkg.builtin.darshan_runtime import DarshanRuntime as BuiltinDarshanRuntime
+
+
+class DarshanRuntime(BuiltinDarshanRuntime):
+    __doc__ = BuiltinDarshanRuntime.__doc__
+
+    version("3.4.0", sha256="7cc88b7c130ec3b574f6b73c63c3c05deec67b1350245de6d39ca91d4cff0842", preferred=True)
```
Comment (matz-e) [inline bluebrain/repo-patches/packages/darshan-runtime/package.py]:
```suggestion
    version("3.4.0", sha256="7cc88b7c130ec3b574f6b73c63c3c05deec67b1350245de6d39ca91d4cff0842")
```

File: bluebrain/repo-patches/packages/darshan-runtime/package.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+    depends_on('autoconf', type='build', when='@3.4.0:')
+    depends_on('automake', type='build', when='@3.4.0:')
+    depends_on('libtool',  type='build', when='@3.4.0:')
+    depends_on('m4',       type='build', when='@3.4.0:')
```
Comment (matz-e) [inline bluebrain/repo-patches/packages/darshan-runtime/package.py]:
```suggestion
    depends_on('autoconf', type='build')
    depends_on('automake', type='build')
    depends_on('libtool',  type='build')
    depends_on('m4',       type='build')
```
You are introducing a bunch of version penalties by adding these requirements to 3.4.0 only. Thus the older 3.3.1 will be preferred, as it has more favorable dependencies :)

File: bluebrain/repo-patches/packages/darshan-runtime/package.py (hunk 1 above)
Comment (matz-e) [reply bluebrain/repo-patches/packages/darshan-runtime/package.py:11]: I'm curious if that works if one expands `"${DARSHAN_LOG_DIR_PATH}"`, too.

**Comment by sergiorg-hpc** [comment]
As per the suggestion of @matz-e, let us keep this PR dedicated to the fix of the log path issue.

**Review by matz-e — DISMISSED** [review]

File: bluebrain/repo-patches/packages/darshan-runtime/package.py [hunk 4]
```diff
@@ -0,0 +1,12 @@
+from spack import *
```
Comment (matz-e) [inline bluebrain/repo-patches/packages/darshan-runtime/package.py]:
```suggestion
```
To keep flake happy

**Review by matz-e — APPROVED** [review]

**Pull request merged**